clippy_config/src/conf.rs

```diff
@@ -642,6 +642,47 @@ define_Conf! {
     ///
     /// Whether to also emit warnings for unsafe blocks with metavariable expansions in **private** macros.
     (warn_unsafe_macro_metavars_in_private_macros: bool = false),
+    /// Lint: FLOAT_CMP
+    ///
+    /// Whether to ignore comparisons to a named constnat
+    ///
+    /// #### Example
+    /// ```no_run
+    /// const VALUE: f64 = 1.0;
+    /// fn is_value(x: f64) -> bool {
+    ///     // Will warn if the config is `false`
+    ///     x == VALUE
+    /// }
+    /// ```
+    (float_cmp_ignore_named_constants: bool = true),
```

Comment:
I wonder if this and the other configurations should really default to `true`. I sadly don't work with any float heavy code to make an informed decision. I'd be in favor of an FCP for these three config values, once everything else has been resolved.

Reply:
Comparisons to named constants should probably take a list of constants. `== PI` probably should be linted for instance. The change detection part is to handle cases like `x + y == x` which is clearly a false positive since adding `y` either changes the value or it doesn't (or NaN happens). Constant comparisons could go either way, but they'll have a high false positive rate.

Reply:
Fair!